### PR TITLE
Fix CI and drop support for py35

### PR DESCRIPTION
**Pull request by adehad**
Making a MR to view the CI after trying out some changes based on the discussions in #896 

Updated the interfaces to match "Jira Server" v2 api - what the docker image is based off (vs "Cloud").
Referring to : https://docs.atlassian.com/software/jira/docs/api/REST/8.13.6/#api/2/ 

FYI: @Addono @ssbarnea - I'd love to drop Py35 support and use cleaner f-strings everywhere, I can imagine these changes would be a major bump anyway due to the using v2 endpoints - so might be worth doing at the same time, but perhaps a separate pull request.

### CODE REVIEW

**Review by studioj — APPROVED** [review]
minor change but it showed warnings ... not really needed for CI :) but good to take along :-)

i saw those warnings too.

Edit:
dammit first time reviewer in a larger github project :-s this comment was ment for the first commit only :-S

i'll try to adjust

File: make_local_jira_user.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from jira import JIRA
```
Comment (studioj) [inline make_local_jira_user.py:1]: nice i wash just cleaning that up too 👍

File: jira/client.py [hunk 2]
```diff
@@ -531,6 +531,29 @@ def __init__(
...
+    def versioned_rest(self):
+        """Return a versioned rest api url"""
+        return "{server}/{api_path}/{api_version}".format(
+            server=self.server,
+            api_path="rest/"+self._options["rest_path"],
+            api_version=self._options["rest_api_version"]
+        )
+
+    @property
+    def latest_rest(self):
```
Comment (studioj) [inline jira/client.py:549]: cleaner, but i need to mention you're moving something private to the public space. 

just need to think about it => more public api => more stuff to maintain :-)

File: jira/client.py [hunk 3]
```diff
@@ -798,7 +821,7 @@ def set_application_property(self, key, value):
         :param value: value to assign to the property
         :type value: str
         """
-        url = self._options["server"] + "/rest/api/latest/application-properties/" + key
+        url = self.latest_rest + "/application-properties/" + key
```
Comment (studioj) [inline jira/client.py:824]: this is a lot clearer 👍

File: jira/client.py [hunk 4]
```diff
@@ -3765,7 +3767,8 @@ def create_project(
             "key": key,
             "projectTypeKey": ptype,
             "projectTemplateKey": template_key,
-            "leadAccountId": assignee,
+            "lead": assignee,
+            # "leadAccountId": assignee,
```
Comment (studioj) [inline jira/client.py:3771]: should this remain here? => i'd say remove the comment if it fixes it

**Review by studioj — COMMENTED** [review]
looks legit 👍 

I'll check out yr branch later today or tomorrow and try to run it locally!

btw thanks a lot for the windows run. I'll try that for sure 👍

File: test.local [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 export CI_JIRA_USER_PASSWORD=jira
 export CI_JIRA_ISSUE=Task
+#sudo docker run -dit -p 2990:2990 --name jira addono/jira-software-standalone
+echo waiting for the jira server to be up and running...
+echo THIS CAN TAKE A WHILE on a vm this can take up to 10 minutes
+start_time="$(date -u +%s)"
+until $(curl -u $CI_JIRA_ADMIN:$CI_JIRA_ADMIN_PASSWORD --output /dev/null --silent --head --fail $CI_JIRA_URL/rest/api/2/permissions); do end_time="$(date -u +%s)";elapsed="$(($end_time-$start_time))"; echo "not running yet after $elapsed seconds $CI_JIRA_URL";sleep 5; done
+pip install -e .
+echo adding user $CI_JIRA_USER to $CI_JIRA_URL
+(python -c "from jira import JIRA;jira_client = JIRA('$CI_JIRA_URL', basic_auth=('$CI_JIRA_ADMIN', '$CI_JIRA_ADMIN_PASSWORD'));a = jira_client.add_user('$CI_JIRA_USER', 'user@example.com', password='$CI_JIRA_USER_PASSWORD') if not jira_client.user('$CI_JIRA_USER') else None" && echo "Created user '$CI_JIRA_USER', or user was already present") || (echo "Failed creating user '$CI_JIRA_USER'" && docker logs --tail 500 jira)
```
Comment (studioj) [inline test.local:16]: maybe your make_py user script can be called here

**Comment by adehad** [comment]
> 
> small remark => should it go to master or first to fix ci?


@studioj I think it can go to master as I've merged the fix_ci stuff already into this branch.


> a quick glance at the CI system shows some missing stuff still
> 
> https://travis-ci.com/github/pycontribs/jira/jobs/500924272
> 
> If i can collaborate here let me know (it's quite late here atm but i'll try to have a look tomorrow )

@studioj definitely appreciate the reviews ! I've just made a quick and dirty fix of all the tests. For whatever reason I cannot find `jirashell` and that test is the only test that fails locally for me.

Likewise a bit late on my side, so will give the reviews a look tomorrow, and tidy up some of the rogue comments and inconsistencies.

I've added you as a collaborator on my fork, so feel free to push stuff to there !

File: jira/client.py (hunk 2 above)
Comment (adehad) [reply jira/client.py]: Yep moved to a private method based of the existing `_get_url()`

File: jira/client.py (hunk 4 above)
Comment (adehad) [reply jira/client.py:3767]: Yep, there's probably a bigger conversation on how we handle Jira Cloud vs Jira Datacenter/Server, but will remove these for now

**Comment by adehad** [comment]
@ssbarnea looks like Travis isn't running any more builds on this PR, any chance you can manually trigger one?

**Comment by ssbarnea** [comment]
Replace travis with gha, travis dead.

**Comment by studioj** [comment]
local run 
image

**Comment by studioj** [comment]
@adehad want me to help in some part?, can i take something up?

**Comment by adehad** [comment]
> 
> 
> @adehad want me to help in some part?, can i take something up?

@studioj Would be great if you are familiar with Github actions and can set something up. I'm not familiar at all so probably won't get very far today

**Comment by studioj** [comment]
haha, if some experience is beter than none then I can try some stuff :-) => i've done one package already on github actions but compared to this one its pretty basic :-)
I'll give it a go 
@ssbarnea I'll try to cover as much as the travis run

**Comment by studioj** [comment]
> I've added you as a collaborator on my fork, so feel free to push stuff to there !

@adehad 

`remote: Permission to adehad/jira.git denied to studioj.
fatal: unable to access 'https://github.com/adehad/jira.git/': The requested URL returned error: 403`

**Comment by studioj** [comment]
i dont have permissions I'll fork from mine but i see you're busy too ...

**Comment by adehad** [comment]
> 
> 
> i dont have permissions I'll fork from mine but i see you're busy too ...

hmm, I sent an invite that may have gone to your mail about giving you access, but feel free to fork off mine

**Comment by studioj** [comment]
```
I'd love to drop Py35 support and use cleaner f-strings everywhere, I can imagine these changes would be a major bump anyway due to the using v2 endpoints - so might be worth doing at the same time, but perhaps a separate pull request.
```
@adehad beter to keep that for a separate PR.

Let's try to focus in this pr on CI improvements only

**Comment by studioj** [comment]
Ok CI is green in this branch.

see here: https://github.com/adehad/jira/actions/runs/803175998

We removed 3.5 support
- that prevents us from locking half of the required packages.

most of the heavy lifting was done by @adehad I'm not that knowledgeable about tox (now a little more)
but we got the Jira Server CI tests up and running AND moved to GitHub Actions.

@ssbarnea I think security wise it might be possible to re add the credentials for testing JiraCloud in GitHub Actions itself.

I don't know if the CI system/test coverage is trustworthy in itself, I think @ssbarnea you can better asses that.

Eager to do some more, but this is a start, ready for getting some green back into this repo ;-)

**Comment by studioj** [comment]
I would add some more intelligent static code analyzer in the CI system too... is there a preference? sourcery/LGTM/Sonar

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
I fixed resolve conflicts in another branch and merged it. I will close this one now.

**Pull request closed without merging**